tests/test_build/test_sidebars_captions.html

```diff
@@ -1,12 +1,17 @@
 <nav aria-label="Main navigation" class="bd-links" id="bd-docs-nav">
  <div class="bd-toc-item active">
+  <p class="caption">
```

Comment:
Hm, a few things from the accessibility angle:
- the caption text could be put directly on the `div`, e.g. `<div aria-label="Section 1"`
- there is also a dedicated HTML5 `<caption>` element, if it's _really_ a caption

Reply:
@bollwyvl the main problem here is that this html is generated by sphinx, so we don't have *direct* control over this snippet. 

If we want to improve this, I think we either need 1) upstream changes to sphinx, or 2) we can adapt the beautifulsoup html object a bit to make some of those targetted changes (as we already do to add some additional classes for styling). But let's consider that out of scope for this PR.